client/src/components/categorical/category.js

```diff
@@ -334,7 +347,11 @@ class Category extends React.Component {
                     rightElement={
                       <Button
                         minimal
-                        disabled={newCategoryText.length === 0}
+                        disabled={
+                          (allCategoryNames.indexOf(newCategoryText) > -1 &&
```

Comment:
I think you should factor this out into a helper function, much like we did with labelError() and labelErrorMessage().